crates/extensions/c8y_mapper_ext/src/operations/handlers/device_profile.rs

```diff
@@ -49,22 +54,50 @@ impl OperationContext {
                 })
             }
             CommandStatus::Successful => {
-                let c8y_target_profile =
-                    MqttMessage::new(sm_topic, set_c8y_profile_target_payload(true)); // Set the target profile as executed
+                let mut messages = Vec::new();
+
+                for operation in command.payload.operations {
+                    let message = match operation.operation.payload() {
+                        OperationPayload::Firmware(firmware) => {
+                            let twin_metadata_topic = self.mqtt_schema.topic_for(
+                                &target.topic_id,
+                                &Channel::EntityTwinData {
+                                    fragment_key: "firmware".to_string(),
+                                },
+                            );
+
+                            MqttMessage::new(&twin_metadata_topic, firmware.to_json())
+                                .with_retain()
+                                .with_qos(QoS::AtLeastOnce)
+                        }
+                        OperationPayload::Software(_) => {
+                            self.request_software_list(&target.topic_id)
+                        }
+                        OperationPayload::Config(config) => MqttMessage::new(
+                            sm_topic,
+                            set_c8y_config_fragment(
+                                &config.config_type,
+                                &config.remote_url.unwrap_or_default(),
```

Comment:
Same comment as the one made in the `config_update.rs`. The local proxy URL prefix must be replaced with the tenant URL.

Reply:
Proxy conversion added in 97ad0bf9de2a4097bf8ae55dfad628602490710f